node/bindnode/repr.go

```diff
@@ -224,19 +226,57 @@ func (w *_nodeRepr) MapIterator() datamodel.MapIterator {
 		itr := (*_node)(w).MapIterator().(*_unionIterator)
 		return (*_unionIteratorRepr)(itr)
 	case nil:
-		return (*_node)(w).MapIterator()
+		switch st := (w.schemaType).(type) {
+		case *schema.TypeMap:
+			return &_mapIteratorRepr{
+				schemaType: st,
+				keysVal:    w.val.FieldByName("Keys"),
+				valuesVal:  w.val.FieldByName("Values"),
+			}
+		default:
+			panic(fmt.Sprintf("TODO: mapitr.repr for typekind %s", w.schemaType.TypeKind()))
+		}
 	default:
 		panic(fmt.Sprintf("TODO: %T", stg))
 	}
 }
 
+type _mapIteratorRepr _mapIterator
```

Comment:
comment to explain why this wrapping and not just calling w's MapIterator is needed would be helpful